### ISSUE TITLE
Wrong behavior for unbound variables in disjunction

### ISSUE THREAD

**Issue by gkastrinis** [souffle-lang/souffle#829]
I have the following simple souffle file that exhibits the following (wrong I believe) behavior.

```
> cat t.dl
.decl A(x:symbol, y:symbol)
.input A
.decl B(x:symbol)
.output B

.decl isA(x:symbol)
isA(x) :- A(_, x).

B(x) :- isA(x), ( A("a", x) ; name = "x1" ).
```

and this input file

```
> cat A.facts
a       x4
c       x5
```

Compiling the above doesn't report any warnings / errors even though variable `name` is not bound out of the disjunction.

The generated binary computes non-intuitive results.

```
> souffle -c t.dl -o foo
> ./foo
> cat B.csv
x4
x5
```

Souffle version: `Version: 1.4.0-444-g72573fbf`

**Comment by azreika** [comment]
Disjunction is just syntactic sugar, so your program is equivalent to:

```
.decl A(x:symbol, y:symbol)
.input A
.decl B(x:symbol)
.output B

.decl isA(x:symbol)
isA(x) :- A(_, x).

B(x) :- isA(x), A("a", x).
B(x) :- isA(x), name = "x1".
```
Focusing on the last rule,

```
B(x) :- isA(x), name = "x1".
```
the variable `name` is bound to the string `"x1"`, and does not affect the values that `x` can be instantiated to. This means that all appearances of `name` will be replaced with `"x1"`. Strictly speaking, `name` does not appear in a grounding atom, so it is ungrounded, but alias resolution (by replacing all appearances of `name` (zero in this case) to `"x1"`) allows us to produce an equivalent grounded rule,

```
B(x) :- isA(x).
```

hence the result.

I agree that there definitely should be a warning for the variable `name`, since it is only used once, and possibly an error when variables in constraints are independent of the body atoms entirely.

**Comment by azreika** [comment]
Fixed by PR #1288.
